lib/src/main/kotlin/com/swmansion/starknet/data/types/Responses.kt

```diff
@@ -396,11 +425,64 @@ data class PendingStateUpdateResponse(
     override val stateDiff: StateDiff,
 ) : StateUpdate()
 
+// TODO: remove SCREAMING_SNAKE_CASE @JsonNames once devnet is updated
+@OptIn(ExperimentalSerializationApi::class)
+@Serializable
+data class ResourceBoundsMapping(
+    @SerialName("l1_gas")
+    @JsonNames("L1_GAS")
+    val l1Gas: ResourceBounds,
+
+    @SerialName("l2_gas")
+    @JsonNames("L2_GAS")
+    val l2Gas: ResourceBounds,
+)
+
+@Serializable
+data class ResourceBounds(
+    @SerialName("max_amount")
+    val maxAmount: Uint64,
+
+    @SerialName("max_price_per_unit")
+    val maxPricePerUnit: Uint128,
+) {
+    companion object {
+        @field:JvmField
+        val ZERO = ResourceBounds(Uint64.ZERO, Uint128.ZERO)
+    }
+
+    fun toMaxFee(): Felt {
+        return maxAmount.value.multiply(maxPricePerUnit.value).toFelt
+    }
```

Comment:
Similar as above, convenience methods that could be changed to extensions, moved to another file or left as is.